### ISSUE TITLE
Neural representation similarities

### ISSUE THREAD

**Issue by domenicoMuscill0** [KevinMusgrave/pytorch-metric-learning#683]
Will you consider to add to this library some similarities among neural network representations? Like CCA or CKA for example.

**Comment by KevinMusgrave** [comment]
Can you provide links to the relevant papers?

I'm wondering if it's out of the scope of this library, or if there is an existing library that serves this purpose well.

**Comment by domenicoMuscill0** [comment]
[CKA](https://arxiv.org/abs/1905.00414) and [CCA](https://arxiv.org/abs/1706.05806) similarities have been implemented by [jayroxis](https://github.com/jayroxis/CKA-similarity) and [moskomule](https://github.com/moskomule/anatome). However the second link is a library with other neural network representation similarities i am not sure it is mantained anymore, but there are tests and pytorch implementation to which compare in the tests of a future implementation.

**Comment by KevinMusgrave** [comment]
Does it make sense to apply CKA and CCA to embeddings?

**Comment by domenicoMuscill0** [comment]
I think it depends on the task. It may be used in some application to NAS or in some other model that could benefit from these similarities

**Comment by KevinMusgrave** [comment]
I mean can they be used as a drop-in replacement for any existing losses, like the contrastive loss?

**Comment by KevinMusgrave** [comment]
Ok let's leave it out for now, unless other people express interest.
